adapters/unruly/unruly.go

```diff
@@ -0,0 +1,180 @@
+package unruly
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type UnrulyAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+func (a *UnrulyAdapter) Name() string {
+	return "unruly"
+}
+
+func (a *UnrulyAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func GetClient(config *adapters.HTTPAdapterConfig) *http.Client {
+	return adapters.NewHTTPAdapter(config).Client
+}
+
+func NewUnrulyAdapter(config *adapters.HTTPAdapterConfig, endpoint string) *UnrulyAdapter {
+	return NewUnrulyBidder(GetClient(config), endpoint)
+}
+
+func NewUnrulyBidder(client *http.Client, endpoint string) *UnrulyAdapter {
+	clientAdapter := &adapters.HTTPAdapter{Client: client}
+
+	return &UnrulyAdapter{
+		http: clientAdapter,
+		URI:  endpoint,
+	}
+}
+
+func (a *UnrulyAdapter) ReplaceImp(imp openrtb.Imp, request *openrtb.BidRequest) *openrtb.BidRequest {
+	reqCopy := *request
+	reqCopy.Imp = append(make([]openrtb.Imp, 0, 1), imp)
+	return &reqCopy
+}
+
+func (a *UnrulyAdapter) CheckImpExtension(request *openrtb.BidRequest) bool {
```

Comment:
is `CheckImpExtension()` used anywhere besides the test to insure it is working?